url7f729337e3_timmolter_Sundial_tgz-pJ8-org_knowm_sundial_SampleRunJ8/src/org/quartz/core/SchedulerFactory.java

```diff
@@ -0,0 +1,188 @@
+ * Copyright 2001-2011 Terracotta Inc. (http://terracotta.org).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.quartz.core;
+
+import org.knowm.sundial.plugins.AnnotationJobTriggerPlugin;
+import org.quartz.QuartzScheduler;
+import org.quartz.exceptions.SchedulerException;
+import org.quartz.plugins.management.ShutdownHookPlugin;
+import org.quartz.plugins.xml.XMLSchedulingDataProcessorPlugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * An implementation of <code>{@link org.quartz.core.SchedulerFactory}</code> that does all of its work of creating a <code>QuartzScheduler</code>
+ * instance.
+ * </p>
+ *
+ * @author James House
+ * @author Anthony Eden
+ * @author Mohammad Rezaei
+ * @author timmolter
+ */
+public class SchedulerFactory {
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
+  private QuartzScheduler quartzScheduler = null;
+
+  private int threadPoolSize = 10; // default size is 10
+  private String packageName = null;
+
+  /**
+   * @param threadPoolSize
+   * @param String packageName
+   * @return Returns a handle to the Scheduler produced by this factory. Initialized with given threadPoolSize and packageName where it looks for
+   *         annotated Job classes
+   * @throws SchedulerException
+   */
+  public Scheduler getScheduler(int threadPoolSize, String packageName) throws SchedulerException {
+
+    this.threadPoolSize = threadPoolSize;
+    this.packageName = packageName;
+
+    return getScheduler();
+  }
+
+  /**
+   * @param threadPoolSize
+   * @return Returns a handle to the Scheduler produced by this factory. Initialized with given threadPoolSize
+   * @throws SchedulerException
+   */
+  public Scheduler getScheduler(int threadPoolSize) throws SchedulerException {
+
+    this.threadPoolSize = threadPoolSize;
+
+    return getScheduler();
+  }
+
+  /**
+   * <p>
+   * Returns a handle to the Scheduler produced by this factory.
+   * </p>
+   * <p>
+   * If one of the <code>initialize</code> methods has not be previously called, then the default (no-arg) <code>initialize()</code> method will be
+   * called by this method.
+   * </p>
+   */
+  public Scheduler getScheduler() throws SchedulerException {
+
+    if (quartzScheduler != null) {
+      return quartzScheduler;
+    }
+
+    return instantiate();
+  }
+
+  private Scheduler instantiate() throws SchedulerException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 58 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 167-186 into a separate method.